tests/resource/valid/definition/function_ret_super.mamba

```diff
@@ -0,0 +1,2 @@
+def some_higher_order(fun: Int -> Int) -> Int => fun(10)
+def fancy() -> Int? => some_higher_order(\x: Int => x * 2)
```

Comment:
Question is, why did this not fail before? We just copied this from another test, but now `fancy` does not have argument `self`, and is not within a class.
Should also add a test with these exact functions but within a class to see if that type checks.